### DISCUSSION TITLE
Construct Column Name For Select By Formatter

### DISCUSSION THREAD

**Discussion by Freydoonk** [MoonStorm/FastCrud#168] [Ideas]
When custom column name is set to entity property, and full select statement is generated by ` OrmConfiguration.GetSqlBuilder<T>().Format` to execute in `Connection.Query`, the generated SQL statement is ignored the actual property name, This will cause the result of Query method wont map to properties.

I can do this by adding new formatter to generate column name with alias.  but I am not sure if it is the only way to do this?
Can I make a pull request?

**Comment by MoonStorm** [comment]
That is weird because I thought internally the library is using the exact sql builder as exposed via ``GetSqlBuilder``.

**Comment by Freydoonk** [comment]
@MoonStorm, When I try to generate a so complex SQL statement, and use `GetSqlBuilderGet.ColumnNameForSelect` for getting each column name, it makes the code more complex. But if there is a formatter like `C` or `TC` to do this, I can simply use `GetSqlBuilderGet.Format`, it makes my code more simple and readable.

For example compare bellow codes:
* `CA` and `TCA` are two new formatters that I added to FastCRUD


```
var sqlBuilder = OrmConfiguration.GetSqlBuilder<Bank>();
var sqlStaement1 = sqlBuilder.Format(
		$@"SELECT
				{sqlBuilder.GetColumnNameForSelect(nameof(Bank.ID))},
				{sqlBuilder.GetColumnNameForSelect(nameof(Bank.BankTitle))}, 
				{sqlBuilder.GetColumnNameForSelect(nameof(Bank.Code))},
				{sqlBuilder.GetColumnNameForSelect(nameof(Bank.Active))},
				{sqlBuilder.GetColumnNameForSelect(nameof(Bank.Description))},
			FROM 
				{nameof(Bank):T}
			WHERE 
				{nameof(Bank.Guid):C}=@param AND
				{nameof(Bank.Deleted):C}=0 AND
				{nameof(Bank.ID):C} NOT IN (SELECT [ID] FROM @ExcludeIDs)
			ORDER BY {nameof(Bank.ID):C} OFFSET 0 ROWS FETCH NEXT 1 ROWS ONLY");
```

-------------

```
var sqlBuilder = OrmConfiguration.GetSqlBuilder<Bank>();
var sqlStaement2 = sqlBuilder.Format(
			$@"SELECT {nameof(Bank.ID):CA},
					{nameof(Bank.BankTitle):CA}, 
					{nameof(Bank.Code):CA},
					{nameof(Bank.Active):CA},
					{nameof(Bank.Description):CA},
				FROM
					{nameof(Bank):T}
				WHERE
					{nameof(Bank.Guid):C}=@param AND
					{nameof(Bank.Deleted):C}=0 AND
					{nameof(Bank.ID):C} NOT IN (SELECT [ID] FROM @ExcludeIDs)
				ORDER BY {nameof(Bank.ID):C} OFFSET 0 ROWS FETCH NEXT 1 ROWS ONLY");
```



* This code is just a simple example, my target is to generate more complex SQL statements
